src/partition/mod.rs

```diff
@@ -992,4 +992,92 @@ impl PartitionHandle {
 
         Ok(())
     }
+
+    /// Removes an item from the partition, leaving behind a weak tombstone.
+    ///
+    /// When a weak tombstone is matched with a single write in a compaction,
+    /// the tombstone will be removed along with the value. If the key was
+    /// overwritten the result of a `remove_single` is undefined.
+    ///
+    /// Only use this remove if it is known that the key has only been written
+    /// to once since its creation or last `remove_single`.
+    ///
+    /// The key may be up to 65536 bytes long.
+    /// Shorter keys result in better performance.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use fjall::{Config, Keyspace, PartitionCreateOptions};
+    /// #
+    /// # let folder = tempfile::tempdir()?;
+    /// # let keyspace = Config::new(folder).open()?;
+    /// # let partition = keyspace.open_partition("default", PartitionCreateOptions::default())?;
+    /// partition.insert("a", "abc")?;
+    ///
+    /// let item = partition.get("a")?.expect("should have item");
+    /// assert_eq!("abc".as_bytes(), &*item);
+    ///
+    /// partition.remove_single("a")?;
+    ///
+    /// let item = partition.get("a")?;
+    /// assert_eq!(None, item);
+    /// #
+    /// # Ok::<(), fjall::Error>(())
+    /// ```
+    ///
+    /// # Errors
+    ///
+    /// Will return `Err` if an IO error occurs.
+    pub fn remove_single<K: Into<UserKey>>(&self, key: K) -> crate::Result<()> {
```

Comment:
Naming the functions `remove_weak` would probably make it more consistent with the "weak tombstone" naming

Reply:
Sure, I'll change them. I didn't originally because, for me, it would've been confusing, given that it's actually a "stronger" or more final removal that involves taking advantage of a weak tombstone. I couldn't think of a good straightforward name that captured what was going on.

Reply:
The reason I named it "weak" tombstone is because it vanishes as soon as it hits a previous value, while a normal tombstone is "invincible" (up to the last level) and shadows everything below it.

Reply:
For sure, it makes sense for the name of the tombstone, it just didn't to me as the name of the `remove` action that when paired with a weak tombstone resulted in the value and the tombstone vanishing.